src/main/kotlin/gg/essential/universal/UDesktop.kt

```diff
@@ -71,13 +71,13 @@ object UDesktop {
     private fun openSystemSpecific(file: String): Boolean {
         return when {
             isLinux -> when {
-                isXdg -> runCommand("xdg-open \"$file\"")
-                isKde -> runCommand("kde-open \"$file\"")
-                isGnome -> runCommand("gnome-open \"$file\"")
-                else -> runCommand("kde-open \"$file\"") || runCommand("gnome-open \"$file\"")
+                isXdg -> runCommand("xdg-open", "\"$file\"")
```

Comment:
Is it still necessary to wrap the file in quotation marks when passing it as an array element?

Reply:
Hm, seems like I missed that when conor sent me the patch.

Reply:
This actually is required for instance if you want a url with spaces in it

Reply:
> This actually is required for instance if you want a url with spaces in it

(or - more particularly a file with spaces, like on windows)

Reply:
I'm pretty sure we do not want the quotes there. There's no reason to have quotes if the file is already its own argument. Quite the opposite actually, cause I don't know of anything that would evaluate the quotes, so this isn't only not necessary, it's incorrect cause the file `"/home/user"` doesn't exist, it's called `/home/user/`.
To further prove my point, let's just try it:
![](https://i.johni0702.de/1YRA6.png)
(yes, I purposefully created that folder beforehand, otherwise it just doesn't open at all)

Reply:
So it seems that it only needs to be changed for linux? afaik it works on mac and windows